utilities/getArticleByTitle/index.ts

```diff
@@ -0,0 +1,29 @@
+import { APPLICATION_JSON, CONTENTFUL_API_ACCESS_TOKEN, CONTENTFUL_API_URL, POST } from '../constants'
+import { ArticleProps } from '../../components/Article'
+
+export default async function getArticleByTitle(title: string): Promise<ArticleProps> {
+  const GET_ARTICLE_BY_TITLE_GRAPHQL_QUERY = `
+    {
+      articleCollection(where: {title: "${title}"}) {
+        items {
+          title
+          content
+        }
+      }
+    }
+  `
+
+  const res = await fetch(CONTENTFUL_API_URL, {
+    method: POST,
+    headers: {
+      'Content-Type': APPLICATION_JSON,
+      Authorization: CONTENTFUL_API_ACCESS_TOKEN,
+    },
+    body: JSON.stringify({ query: GET_ARTICLE_BY_TITLE_GRAPHQL_QUERY }),
+  })
+  const { data } = await res.json()
+  // articleCollection can only contains zero or one article. Since title is an unique field.
+  const article: ArticleProps = data.articleCollection.items[0]
```

Comment:
Wonder if there's a query where you can get one or none results for a unique key?

Reply:
One or none is always possible -- just misspell the key. So we need to allow for that.

Reply:
Also, this is a better way of doing the above, I think:

```typescript
const [pageContent]: PageContentProps[] = data.pageContent.items
```

Reply:
@natebowang does Contentful return an items array even if you're searching for a single item by a unique key?

Reply:
I will work on that. I think Contentful should have a way to just query a record. I am new to GraphQL and will have some research on that. 
Is generate GraphQL query by a query builder better than just strings? Like [this one](https://www.npmjs.com/package/gql-query-builder).

Reply:
@reubenberghan may have his own opinion, but I think not. Write GraphQL queries in GraphQL. Write CSS in CSS. Write HTML, to the extent possible, in HTML (or JSX, which is mostly equivalent). Devs who have a hammer and want everything to be a nail frustrate the hell out of me. USE THE RIGHT TOOL FOR THE JOB, FOOLS. (Did you see my rant about CSS-in-JS? Same applies.)

Here is what it looks like with `graphql-tag`:

```typescript
import gql from 'graphql-tag'

export const teamsByMemberIdQuery = gql`
  query TeamsByMemberId($memberId: ID!) {
    teamsByMemberId(memberId: $memberId) {
      displayName
      id
      createdAt
      updatedAt
    }
  }
`
```

Definitely, learning GraphQL well should be part of this exercise.